From 2c5d99b57ba613de47f99b356fc88e964b878335 Mon Sep 17 00:00:00 2001
From: Yangbo Lu <yangbo.lu@nxp.com>
Date: Thu, 16 Jan 2020 22:58:24 +0200
Subject: [PATCH 1/2] net: mscc: ocelot: avoid incorrect consuming in skbs list

[ Upstream commit fc62c0948986b7aa6b2871a450a4469e35c5f9bc ]

Break the matching loop when find the matching skb for TX timestamp.
This is to avoid consuming more skbs incorrectly. The timestamp ID
is from 0 to 3 while the FIFO could support 128 timestamps at most.

Signed-off-by: Yangbo Lu <yangbo.lu@nxp.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
Signed-off-by: Vladimir Oltean <vladimir.oltean@nxp.com>
---
 drivers/net/ethernet/mscc/felix_board.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/net/ethernet/mscc/felix_board.c b/drivers/net/ethernet/mscc/felix_board.c
index a3f9d473cd85..1bce32f5a484 100644
--- a/drivers/net/ethernet/mscc/felix_board.c
+++ b/drivers/net/ethernet/mscc/felix_board.c
@@ -494,6 +494,7 @@ static void felix_tx_clean(struct ocelot *ocelot)
 
 			list_del(pos);
 			devm_kfree(ocelot->dev, entry);
+			break;
 		}
 
 		if (likely(skb)) {
-- 
2.17.1

